### PR TITLE
Create a new endpoint to return collection level dependencies

### PR DESCRIPTION
**Pull request by bmclaughlin**
Create a new ``collections/ns/name/versions/version/dependencies/`` endpoint that returns collection level dependencies.

https://pulp.plan.io/issues/8132
closes #8132

### CODE REVIEW

**Review by bmbouter — CHANGES_REQUESTED** [review]
Please add tests that drive this endpoint and assert on its results.

**Comment by bmbouter** [comment]
We plan to release 0.7 on Feb 9th so this needs to be merged by Feb 8th if you want it in that release.

File: CHANGES/8132.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Create a new ``collections/ns/name/versions/version/dependencies/``endpoint that returns collection level dependencies.
```
Comment (webknjaz) [inline CHANGES/8132.feature:1]:
```suggestion
Create a new ``collections/ns/name/versions/version/dependencies/`` endpoint that returns all collection transitive dependencies.
```

**Comment by bmclaughlin** [comment]
Closing in favor of the implementation at https://github.com/ansible/ansible/pull/72591.  :+1:

**Comment by jborean93** [comment]
I agree with the comments made here there are a lot of things that need to happen on the client side and I think that having the API resolve the dependencies will only solve some of those cases and not all making it hard to implement this new API in the cli. What might be better is exposing an API were a client can get a manifest of data in 1 go that includes things like

* All the collections
* Their versions
* Collection info like
    * Dependencies
    * Plugin information

So instead of making tens to hundreds of API calls to get the collection information the client only gets this manifest info and now it has all the data it needs to resolve the collections itself in whatever way it wishes. It can regularly get the updated manifest. I'm no API expert but it seems like tools like `dnf`/`apt` provide something like this to make their tools more efficient.

**Comment by bmbouter** [comment]
@jborean93 fyi check out these new galaxy V3 apis that just got added for exactly that reason: https://github.com/pulp/pulp_ansible/pull/488/files

**Comment by jborean93** [comment]
Fantastic that should help the cli a lot more, any chance of it being done to the `galaxy.ansible` v2 endpoint?

**Comment by fao89** [comment]
@jborean93 do you mean v2 endpoints on pulp side? 
If so, yes we can do it, but if you mean https://galaxy.ansible.com/api/v2/ we can't do it, since it is not pulp code -  https://github.com/ansible/galaxy

**Comment by jborean93** [comment]
I mean the latter, the bulk of the `ansible-galaxy` cli users will be people talking to galaxy.ansible.com so even if we improve the experience on Automation Hub it won't really help the tool overall as Galaxy is still using the slower API. Understandable you can't do much because it doesn't use Pulp but maybe it's a topic to bring up with the Galaxy team to get it to use the v2 Pulp API.

**Pull request closed without merging**